module/vision/BallDetector/README.md

```diff
@@ -1,13 +1,40 @@
-BallDetector
-============
+# BallDetector
 
 ## Description
 
+The BallDetector module performs post-processing on the output of the VisualMesh to determine where balls are in the image, if any. If there are no balls, it will emit an empty `Balls` message.
+
+The BallDetector first tries to create balls, and then removes candidate balls if they fail to pass a set of checks.
+
+Balls are created by clustering neighbouring Visual Mesh points that have a high enough confidence that they are a ball point.
+
+Each cluster is checked to see if
+
+1. It is below the green horizon
+2. It has a high enough degree of fit
```

Comment:
Degree of fit to a ... circle?

Reply:
cone?

Reply:
I mean we are fitting a cone, but this check in particular is really measuring how circular it is